Backend/ManagerLayer/AccessControl/AuthorizationManager.cs

```diff
@@ -50,56 +53,39 @@ public Session CreateSession(DatabaseContext _db, User user)
            
         }
 
-        public string ValidateAndUpdateSession(string token)
+        public Session ValidateAndUpdateSession(DatabaseContext _db, string token)
         {
-            using (var _db = CreateDbContext())
-            {
-                Session response = _sessionService.ValidateSession(_db, token);
+            Session response = _sessionService.ValidateSession(_db, token);
 
-                if(response != null)
-                {
-                    response = _sessionService.UpdateSession(_db, response);
-                }
-                else
-                {
-                    return null;
-                }
-
-                try

```

Comment:
All error handling has moved to the controllers. _db.saveChanges() is now called in the controller instead of here.